### PR TITLE
add tracking for dcr table of contents

### PR DESCRIPTION
**Pull request by marjisound**
## What does this change?
This PR adds the following trackings for the TableOdContents component:
- view tracking for the `TableOfContents` component
- click tracking for when the table is expanded with link name as `table-of-contents-expand`
- click tracking for when the table is closed with link name as `table-of-contents-close`
- click tracking for every item in the table with link name pattern as `table-of-contents-<index>-<id-of-item>`

Note: To be able to add tracking for when the table is expanded/closed, we need to make the component importable and use Island, in order to have state in the component.

## Why?
Needed for the data insight team

## Test
Tested this locally using the Ophan tester tool

### CODE REVIEW

**Review by abeddow91 — APPROVED** [review]
LGTM!

File: dotcom-rendering/src/web/components/TableOfContents.tsx [hunk 1]
```diff
@@ -79,8 +84,12 @@ export const TableOfContents = ({ tableOfContents }: Props) => {
 			</summary>
 
 			<ul>
-				{tableOfContents.map((item) => (
-					<li key={item.id} css={listItemStyles}>
+				{tableOfContents.map((item, index) => (
+					<li
+						key={item.id}
+						css={listItemStyles}
+						data-link-name={`toc-item-${index}-${item.id}`}
```
Comment (jamesgorrie) [inline dotcom-rendering/src/web/components/TableOfContents.tsx:91]: Just a thought of using the `toc` namespace but `table-of-contents` parent name might be a little cognitive load for analysts

i.e. `table-of-contents-${index}-{item.id}` or calling the parent `toc` standardises it a little.

**Review by jamesgorrie — APPROVED** [review]

File: dotcom-rendering/src/web/components/TableOfContents.tsx (hunk 1 above)
Comment (marjisound) [reply dotcom-rendering/src/web/components/TableOfContents.tsx:102]: Great idea, I changed them all to have `table-of-contents`

**Comment by marjisound** [comment]
[enable table of contents on web and app](https://trello.com/c/rhNXYQJl/587-enable-table-of-contents-on-web-and-app)

**Pull request merged**